security/pkg/pki/ca/ca.go

```diff
@@ -210,23 +226,85 @@ func NewSelfSignedIstioCAOptions(ctx context.Context,
 	return caOpts, err
 }
 
-func loadSelfSignedCaSecret(client corev1.CoreV1Interface, namespace string, caCertName string, rootCertFile string, caOpts *IstioCAOptions) error {
+// loadCASecrets loads the original istio-ca-secret Secret from K8S.
+// To simplify - this secret should be used for both self-generated and user defined intermediary certs.
```

Comment:
The self-signed secret today may hold the user defined additional roots. Few people know and use this - but the code has supported it for a long time. Not planning to change that since it may break the few users who figured it out. 

I am not sure what happens on the self-signed cert rotation - but I don't think we override the roots.